lib/workers/repository/process/write.ts

```diff
@@ -34,8 +35,15 @@ export async function writeUpdates(
 
   for (const branch of branches) {
     addMeta({ branch: branch.branchName });
-    const branchExisted = branchExists(branch.branchName);
-    const res = await processBranch(branch);
+    let branchExisted;
+    let res;
+    if (
+      GlobalConfig.get('dryRun') !== 'extract' &&
+      GlobalConfig.get('dryRun') !== 'lookup'
+    ) {
+      branchExisted = branchExists(branch.branchName);
+      res = await processBranch(branch);
+    }
```

Comment:
Again here, it would be better to add a "if dryRun===extract|lookup then return" before doing any branch work